gradle/scripts/yaml.gradle

```diff
@@ -44,6 +63,31 @@ def validateSchema(yamlFile, jsonSchemaFile) {
     }
 }
 
+ext.validateMapsYamlLinks = { yamlFile ->
+    def yaml = new Yaml()
+    def maps = yaml.load(yamlFile.text)
+    def imageUrls = maps . collect { it['img'] } . findAll { it != null }
+
+    def clientConfig = new DefaultAsyncHttpClientConfig.Builder()
+            .setFollowRedirect(true)
+            .build()
+    def client = new DefaultAsyncHttpClient(clientConfig)
+    try {
+        def tasks = imageUrls.collect { client.prepareHead(it).execute() }
+
+        def task = null
+        while ((task = poll(tasks, { it.done })) != null) {
+            def response = task.get()
+            logger.info("Map thumbnail image status ($response.uri): $response.statusCode")
+            if (response.statusCode != HttpStatus.SC_OK) {
+                throw new GradleException("$response.uri: map thumbnail image not available ($response.statusCode)")
+            }
+        }
+    } finally {
+        client.close()
```

Comment:
In-flight requests will be shut down here if, for example, we fail the build above.